app/src/main/java/org/jellyfin/mobile/player/PlayerActivity.kt

```diff
@@ -232,9 +247,77 @@ class PlayerActivity : AppCompatActivity() {
                 }
                 return true
             }
+
+            override fun onScroll(
+                e1: MotionEvent?,
+                e2: MotionEvent?,
+                distanceX: Float,
+                distanceY: Float
+            ): Boolean {
+                if(e1 == null)
+                    return false
+
+                if(abs(distanceY / distanceX) < 2)
+                    return true
+
+                val viewCenterX = playerView.measuredWidth / 2
+
+                // Distance to swipe to go from min to max
+                val distanceFull = playerView.measuredHeight
+
+                val ratioChange = distanceY / distanceFull
+
+                if(e1.x.toInt()>viewCenterX){
+                    // On right so changing volume
+                    val currentVolume = audioManager.getStreamVolume(AudioManager.STREAM_MUSIC)
+                    if(counter == -1f) counter = currentVolume.toFloat()
+
+                    val maxVolume = audioManager.getStreamMaxVolume(AudioManager.STREAM_MUSIC)
+                    val change = ratioChange * maxVolume
+                    counter += change
+
+                    val toSet = counter.toInt().coerceIn(0,maxVolume)
+                    audioManager.setStreamVolume(AudioManager.STREAM_MUSIC,toSet,0)
+
+                    overlayBrightnessVolumeImage.setImageResource(R.drawable.ic_volume_white_84dp)
+                    overlayBrightnessVolumeProgressBar.max = maxVolume
+                    overlayBrightnessVolumeProgressBar.progress = toSet
+                }
+                else{
+                    // On left so changing brightness
+                    val windowLayoutParams = window.attributes
+                    if(counter == -1f){
+                        counter = windowLayoutParams.screenBrightness
+                        if(counter < 0f)
+                            counter = android.provider.Settings.System.getFloat(
```

Comment:
@Maxr1998 @nielsvanvelzen This line has a few problems. According to the docs, it should give value between 0 and 255 but on my phone (which has MIUI) it is goes over 255.
Worst case scenario, it jumps to full brightness first time the gesture is used.
I can disable it but imo the utility it offers is worth dealing with the bug.